diff -Naur old/globaliser/lexer.l new/globaliser/lexer.l
--- old/globaliser/lexer.l	2012-05-08 22:30:13.000000000 +0200
+++ new/globaliser/lexer.l	2020-09-03 12:58:14.503468124 +0200
@@ -25,6 +25,7 @@
 %{
 #include <ctype.h>
 #include <stdio.h>
+#include <string.h>
 #include <sstream>
 
 #include "parser.tab.hh"
diff -Naur old/globaliser/node.cc new/globaliser/node.cc
--- old/globaliser/node.cc	2012-05-08 22:30:13.000000000 +0200
+++ new/globaliser/node.cc	2020-09-03 12:58:14.503468124 +0200
@@ -7,7 +7,9 @@
 #include "../sim/num_stacks.h"
 #include "handle_global.h"
 
-using namespace std;
+using std::string;
+using std::stringstream;
+using std::list;
 
 // Exported from the lexer
 extern string ws_text;
diff -Naur old/globaliser/node.h new/globaliser/node.h
--- old/globaliser/node.h	2012-05-08 22:30:13.000000000 +0200
+++ new/globaliser/node.h	2020-09-03 12:58:14.503468124 +0200
@@ -19,8 +19,8 @@
 
 #include <string>
 #include <list>
-
-using namespace std;
+using std::string;
+using std::list;
 
 struct node_t
 {
diff -Naur old/globaliser/parser.yc new/globaliser/parser.yc
--- old/globaliser/parser.yc	2012-05-08 22:30:13.000000000 +0200
+++ new/globaliser/parser.yc	2020-09-03 12:58:14.503468124 +0200
@@ -38,7 +38,6 @@
 #include <algorithm>
 #include <assert.h>
 #include "../sim/num_stacks.h"
-using namespace std;
 
 #define YYERROR_VERBOSE
 #define YYSTYPE struct node_t *
diff -Naur old/linux-2.6.18/include/linux/compiler.h new/linux-2.6.18/include/linux/compiler.h
--- old/linux-2.6.18/include/linux/compiler.h	2012-05-08 22:30:13.000000000 +0200
+++ new/linux-2.6.18/include/linux/compiler.h	2020-09-03 12:59:10.966624453 +0200
@@ -36,8 +36,20 @@
 
 #ifdef __KERNEL__
 
-#if __GNUC__ > 4
+#if __GNUC__ > 10
 #error no compiler-gcc.h file for this gcc version
+#elif __GNUC__ == 10
+# include <linux/compiler-gcc4.h>
+#elif __GNUC__ == 9
+# include <linux/compiler-gcc4.h>
+#elif __GNUC__ == 8
+# include <linux/compiler-gcc4.h>
+#elif __GNUC__ == 7
+# include <linux/compiler-gcc4.h>
+#elif __GNUC__ == 6
+# include <linux/compiler-gcc4.h>
+#elif __GNUC__ == 5
+# include <linux/compiler-gcc4.h>
 #elif __GNUC__ == 4
 # include <linux/compiler-gcc4.h>
 #elif __GNUC__ == 3
